components/FinancialSummary/BarChartComponent.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { useEffect, useState } from 'react';
+import { Bar, BarChart, CartesianGrid, Legend, Tooltip, YAxis } from 'recharts';
+
+import type { CustomTooltipProps, ExpenseItem, Expenses, ExpensesLinkItem } from '@/types/FinancialSummary/BarChartComponent';
+
+import ExpensesData from '../../config/finance/json-data/2024/Expenses.json';
+import ExpensesLinkData from '../../config/finance/json-data/2024/ExpensesLink.json';
+import { getUniqueCategories } from '../../utils/getUniqueCategories';
+
+/**
+ * CustomTooltip component displays custom tooltip for BarChart.
+ * @param {CustomTooltipProps} props - Props for CustomTooltip component.
+ * @returns {JSX.Element} CustomTooltip component.
+ */
+const CustomTooltip: React.FC<CustomTooltipProps> = ({ active, payload }) => {
+  if (active && payload && payload.length) {
+    const data = payload[0].payload;
+
+    return (
+      <div className='bg-opacity-90/90 rounded-md border border-gray-300 bg-white p-2 shadow-md'>
+        <p className='text-14 mb-1 font-bold'>{data.Category}</p>
+        <p className='text-12 text-gray-900'>${data.Amount.toFixed(2)}</p>
+        <p>Click the bar to learn more</p>
+      </div>
+    );
+  }
+
+  return null;
+};
```

Comment:
Use separate files to write these components. We shouldn't have multiple React components inside single file. Same goes for `Card` and `ExpensesCard`.